nixos/tests/installer.nix

```diff
@@ -123,6 +123,9 @@ let
       $machine->succeed("type -tP ps | tee /dev/stderr") =~ /.nix-profile/
           or die "nix-env failed";
 
+      # This should do nothing
+      $machine->succeed("nixos-rebuild switch >&2");
```

Comment:
FYI, `nixos-rebuild switch` does not do nothing.  In all cases, even if this is the same as the previous system, it will still create a new generation to the system profile, update the symbolic link, and run the activation script.

Maybe, what you should be testing is the the `current-system` still targets the same path in the `/nix/store`, to check the the output is the same.